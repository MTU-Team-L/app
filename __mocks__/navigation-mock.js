import {jest} from 'jest-without-globals';

export default {
  setOptions: jest.fn(),
  navigate: jest.fn(),
  isFocused: jest.fn()
};
