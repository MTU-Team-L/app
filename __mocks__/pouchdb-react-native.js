import {jest} from 'jest-without-globals';
import seeds from './seed-data.json';

export const mockPouchDB = jest.fn();

export const put = jest.fn();

const mock = mockPouchDB.mockImplementation(db => {
  return {
    allDocs: jest.fn().mockImplementation(() => Promise.resolve(seeds[db])),
    put,
    changes: jest.fn().mockImplementation(() => ({on: jest.fn()}))
  };
});

export default mock;
