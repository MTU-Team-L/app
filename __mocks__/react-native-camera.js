import React from 'react';
import {View} from 'react-native';

const sampleResponse = [{bounds: {origin: {y: 127.37640380859375, x: 79.48877716064453}, size: {height: 14.498126983642578, width: 80.42393493652344}}, type: 'block', value: 'Jace Beleren', components: [{components: [{bounds: {origin: {y: 127.37640380859375, x: 79.48877716064453}, size: {height: 14.498126983642578, width: 34.60099792480469}}, value: 'Jace', type: 'element'}, {bounds: {origin: {y: 130.48313903808594, x: 119.70074462890625}, size: {height: 9.32022476196289, width: 40.21196746826172}}, value: 'Beleren', type: 'element'}], value: 'Jace Beleren', type: 'line', bounds: {size: {width: 80.42393493652344, height: 14.498126983642578}, origin: {x: 79.48877716064453, y: 127.37640380859375}}}]}, {bounds: {origin: {y: 311.7097473144531, x: 101.93266296386719}, size: {height: 8.28464412689209, width: 54.23939895629883}}, type: 'block', value: 'Planeswalker', components: [{components: [{bounds: {origin: {y: 311.7097473144531, x: 101.93266296386719}, size: {height: 8.28464412689209, width: 54.23939895629883}}, value: 'Planeswalker', type: 'element'}], value: 'Planeswalker', type: 'line', bounds: {size: {width: 54.23939895629883, height: 8.28464412689209}, origin: {x: 101.93266296386719, y: 311.7097473144531}}}]}, {bounds: {origin: {y: 312.7453308105469, x: 171.13465881347656}, size: {height: 8.28464412689209, width: 16.832916259765625}}, type: 'block', value: 'Jace', components: [{components: [{bounds: {origin: {y: 312.7453308105469, x: 171.13465881347656}, size: {height: 8.28464412689209, width: 16.832916259765625}}, value: 'Jace', type: 'element'}], value: 'Jace', type: 'line', bounds: {size: {width: 16.832916259765625, height: 8.28464412689209}, origin: {x: 171.13465881347656, y: 312.7453308105469}}}]}, {bounds: {origin: {y: 332.4213562011719, x: 119.70074462890625}, size: {height: 13.462546348571777, width: 90.71072387695312}}, type: 'block', value: 'ach player drawy& card', components: [{components: [{bounds: {origin: {y: 335.528076171875, x: 119.70074462890625}, size: {height: 9.32022476196289, width: 17.76807975769043}}, value: 'ach', type: 'element'}, {bounds: {origin: {y: 333.4569396972656, x: 139.33914184570312}, size: {height: 11.391386032104492, width: 23.379051208496094}}, value: 'player', type: 'element'}, {bounds: {origin: {y: 333.4569396972656, x: 163.6533660888672}, size: {height: 11.391386032104492, width: 29.925186157226562}}, value: 'drawy&', type: 'element'}, {bounds: {origin: {y: 332.4213562011719, x: 194.51370239257812}, size: {height: 12.426965713500977, width: 15.897754669189453}}, value: 'card', type: 'element'}], value: 'ach player drawy& card', type: 'line', bounds: {size: {width: 90.71072387695312, height: 13.462546348571777}, origin: {x: 119.70074462890625, y: 332.4213562011719}}}]}, {bounds: {origin: {y: 363.48876953125, x: 92.58104705810547}, size: {height: 9.32022476196289, width: 124.37655639648438}}, type: 'block', value: 'CITarget player draws a cand.', components: [{components: [{bounds: {origin: {y: 363.48876953125, x: 92.58104705810547}, size: {height: 9.32022476196289, width: 49.56359100341797}}, value: 'CITarget', type: 'element'}, {bounds: {origin: {y: 365.5599365234375, x: 145.88528442382812}, size: {height: 7.249063491821289, width: 20.573566436767578}}, value: 'player', type: 'element'}, {bounds: {origin: {y: 365.5599365234375, x: 170.19949340820312}, size: {height: 5.177902698516846, width: 19.638402938842773}}, value: 'draws', type: 'element'}, {bounds: {origin: {y: 363.48876953125, x: 191.70822143554688}, size: {height: 9.32022476196289, width: 6.5461344718933105}}, value: 'a', type: 'element'}, {bounds: {origin: {y: 363.48876953125, x: 200.1246795654297}, size: {height: 9.32022476196289, width: 16.832916259765625}}, value: 'cand.', type: 'element'}], value: 'CITarget player draws a cand.', type: 'line', bounds: {size: {width: 124.37655639648438, height: 9.32022476196289}, origin: {x: 92.58104705810547, y: 363.48876953125}}}]}, {bounds: {origin: {y: 380.05804443359375, x: 120.63591003417969}, size: {height: 10.355805397033691, width: 141.20947265625}}, type: 'block', value: 'Target player puts the top wenty cards', components: [{components: [{bounds: {origin: {y: 382.12921142578125, x: 120.63591003417969}, size: {height: 8.28464412689209, width: 22.443889617919922}}, value: 'Target', type: 'element'}, {bounds: {origin: {y: 381.0936279296875, x: 146.82044982910156}, size: {height: 9.32022476196289, width: 20.573566436767578}}, value: 'player', type: 'element'}, {bounds: {origin: {y: 383.164794921875, x: 170.19949340820312}, size: {height: 6.213482856750488, width: 14.962593078613281}}, value: 'puts', type: 'element'}, {bounds: {origin: {y: 381.0936279296875, x: 188.90274047851562}, size: {height: 7.249063491821289, width: 10.286783218383789}}, value: 'the', type: 'element'}, {bounds: {origin: {y: 383.164794921875, x: 201.99501037597656}, size: {height: 6.213482856750488, width: 12.157106399536133}}, value: 'top', type: 'element'}, {bounds: {origin: {y: 382.12921142578125, x: 216.9575958251953}, size: {height: 7.249063491821289, width: 23.379051208496094}}, value: 'wenty', type: 'element'}, {bounds: {origin: {y: 381.0936279296875, x: 243.1421356201172}, size: {height: 7.249063491821289, width: 18.7032413482666}}, value: 'cards', type: 'element'}], value: 'Target player puts the top wenty cards', type: 'line', bounds: {size: {width: 141.20947265625, height: 10.355805397033691}, origin: {x: 120.63591003417969, y: 380.05804443359375}}}]}, {bounds: {origin: {y: 390.4138488769531, x: 93.51620483398438}, size: {height: 8.28464412689209, width: 149.6259307861328}}, type: 'block', value: 'O:of his or her library into his or her', components: [{components: [{bounds: {origin: {y: 390.4138488769531, x: 93.51620483398438}, size: {height: 8.28464412689209, width: 34.60099792480469}}, value: 'O:of', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 130.9226837158203}, size: {height: 7.249063491821289, width: 8.416458129882812}}, value: 'his', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 143.07980346679688}, size: {height: 7.249063491821289, width: 6.5461344718933105}}, value: 'or', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 153.3665771484375}, size: {height: 7.249063491821289, width: 10.286783218383789}}, value: 'her', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 167.39401245117188}, size: {height: 8.28464412689209, width: 22.443889617919922}}, value: 'library', type: 'element'}, {bounds: {origin: {y: 392.4850158691406, x: 193.57855224609375}, size: {height: 4.142322063446045, width: 13.092268943786621}}, value: 'into', type: 'element'}, {bounds: {origin: {y: 390.4138488769531, x: 209.47630310058594}, size: {height: 6.213482856750488, width: 9.3516206741333}}, value: 'his', type: 'element'}, {bounds: {origin: {y: 392.4850158691406, x: 222.56857299804688}, size: {height: 4.142322063446045, width: 6.5461344718933105}}, value: 'or', type: 'element'}, {bounds: {origin: {y: 391.4494323730469, x: 232.85536193847656}, size: {height: 5.177902698516846, width: 10.286783218383789}}, value: 'her', type: 'element'}], value: 'O:of his or her library into his or her', type: 'line', bounds: {size: {width: 149.6259307861328, height: 8.28464412689209}, origin: {x: 93.51620483398438, y: 390.4138488769531}}}]}, {bounds: {origin: {y: 400.7696533203125, x: 117.8304214477539}, size: {height: 7.249063491821289, width: 37.4064826965332}}, type: 'block', value: 'graveyard.', components: [{components: [{bounds: {origin: {y: 400.7696533203125, x: 117.8304214477539}, size: {height: 7.249063491821289, width: 37.4064826965332}}, value: 'graveyard.', type: 'element'}], value: 'graveyard.', type: 'line', bounds: {size: {width: 37.4064826965332, height: 7.249063491821289}, origin: {x: 117.8304214477539, y: 400.7696533203125}}}]}, {bounds: {origin: {y: 412.1610412597656, x: 160.84786987304688}, size: {height: 10.355805397033691, width: 50.49875259399414}}, type: 'block', value: 'Ak Bril', components: [{components: [{bounds: {origin: {y: 414.2322082519531, x: 160.84786987304688}, size: {height: 8.28464412689209, width: 25.24937629699707}}, value: 'Ak', type: 'element'}, {bounds: {origin: {y: 412.1610412597656, x: 193.57855224609375}, size: {height: 7.249063491821289, width: 17.76807975769043}}, value: 'Bril', type: 'element'}], value: 'Ak Bril', type: 'line', bounds: {size: {width: 50.49875259399414, height: 10.355805397033691}, origin: {x: 160.84786987304688, y: 412.1610412597656}}}]}];

// eslint-disable-next-line import/no-mutable-exports
let triggerTextRecognition = () => {};

const RNCamera = ({onTextRecognized}) => {
  triggerTextRecognition = async () => {
    await onTextRecognized({textBlocks: sampleResponse});
  };

  return <View/>;
};

RNCamera.Constants = {
  Type: {
    back: 'mock-camera-back'
  },
  FlashMode: {
    on: 'mock-on'
  }
};

export {RNCamera, triggerTextRecognition};
