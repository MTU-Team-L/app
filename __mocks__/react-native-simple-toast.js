import {jest} from 'jest-without-globals';

const mock = {
  SHORT: 'short',
  LONG: 'long',
  show: jest.fn()
};

module.exports = mock;
