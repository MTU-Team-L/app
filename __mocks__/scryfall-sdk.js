import {jest} from 'jest-without-globals';
import {scryfallCard} from './seed-data.json';

const Cards = {
  byName: () => Promise.resolve(scryfallCard),
  search: jest.fn()
};

export {Cards};
