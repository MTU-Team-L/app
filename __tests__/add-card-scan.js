import React from 'react';
import {test, expect, jest} from 'jest-without-globals';
import {render} from 'react-native-testing-library';
import {triggerTextRecognition} from '../__mocks__/react-native-camera';
import * as scryfall from '../__mocks__/scryfall-sdk';
import {put} from '../__mocks__/pouchdb-core-react-native';
import navigationMock from '../__mocks__/navigation-mock';
import AddCardScan from '../src/scenes/add-card-scan';

const mockCardResponse = {object: 'card', id: '97956927-135f-47b0-8aef-cc63889eab5a', oracle_id: 'cc2b6e58-2d6f-495e-95f3-9496105c4cba', multiverse_ids: [393851], tcgplayer_id: 93796, name: 'Jace Beleren', lang: 'en', released_at: '2014-12-05', uri: 'https://api.scryfall.com/cards/97956927-135f-47b0-8aef-cc63889eab5a', scryfall_uri: 'https://scryfall.com/card/jvc/1/jace-beleren?utm_source=api', layout: 'normal', highres_image: true, image_uris: {small: 'https://img.scryfall.com/cards/small/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464', normal: 'https://img.scryfall.com/cards/normal/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464', large: 'https://img.scryfall.com/cards/large/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464', png: 'https://img.scryfall.com/cards/png/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.png?1562378464', art_crop: 'https://img.scryfall.com/cards/art_crop/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464', border_crop: 'https://img.scryfall.com/cards/border_crop/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464'}, mana_cost: '{1}{U}{U}', cmc: 3, type_line: 'Legendary Planeswalker — Jace', oracle_text: '+2: Each player draws a card.\n−1: Target player draws a card.\n−10: Target player puts the top twenty cards of their library into their graveyard.', loyalty: '3', colors: ['U'], color_identity: ['U'], legalities: {standard: 'not_legal', future: 'not_legal', historic: 'not_legal', pioneer: 'not_legal', modern: 'legal', legacy: 'legal', pauper: 'not_legal', vintage: 'legal', penny: 'not_legal', commander: 'legal', brawl: 'not_legal', duel: 'legal', oldschool: 'not_legal'}, games: ['paper'], reserved: false, foil: true, nonfoil: false, oversized: false, promo: false, reprint: true, variation: false, set: 'jvc', set_name: 'Duel Decks Anthology: Jace vs. Chandra', set_type: 'duel_deck', set_uri: 'https://api.scryfall.com/sets/37832684-2fe0-4b06-842b-eec06e5a17cb', set_search_uri: 'https://api.scryfall.com/cards/search?order=set&q=e%3Ajvc&unique=prints', scryfall_set_uri: 'https://scryfall.com/sets/jvc?utm_source=api', rulings_uri: 'https://api.scryfall.com/cards/97956927-135f-47b0-8aef-cc63889eab5a/rulings', prints_search_uri: 'https://api.scryfall.com/cards/search?order=released&q=oracleid%3Acc2b6e58-2d6f-495e-95f3-9496105c4cba&unique=prints', collector_number: '1', digital: false, rarity: 'mythic', card_back_id: '0aeebaf5-8c7d-4636-9e82-8c27447861f7', artist: 'Kev Walker', artist_ids: ['f366a0ee-a0cd-466d-ba6a-90058c7a31a6'], illustration_id: '00146337-02df-4b1e-976b-8d64af966dd8', border_color: 'black', frame: '2015', full_art: false, textless: false, booster: false, story_spotlight: false, edhrec_rank: 884, prices: {usd: null, usd_foil: '3.94', eur: null, tix: null}, related_uris: {gatherer: 'https://gatherer.wizards.com/Pages/Card/Details.aspx?multiverseid=393851', tcgplayer_decks: 'https://decks.tcgplayer.com/magic/deck/search?contains=Jace+Beleren&page=1&partner=Scryfall&utm_campaign=affiliate&utm_medium=api&utm_source=scryfall', edhrec: 'https://edhrec.com/route/?cc=Jace+Beleren', mtgtop8: 'https://mtgtop8.com/search?MD_check=1&SB_check=1&cards=Jace+Beleren'}, purchase_uris: {tcgplayer: 'https://shop.tcgplayer.com/product/productsearch?id=93796&partner=Scryfall&utm_campaign=affiliate&utm_medium=api&utm_source=scryfall', cardmarket: 'https://www.cardmarket.com/en/Magic/Products/Search?referrer=scryfall&searchString=Jace+Beleren&utm_campaign=card_prices&utm_medium=text&utm_source=scryfall', cardhoarder: 'https://www.cardhoarder.com/cards?affiliate_id=scryfall&data%5Bsearch%5D=Jace+Beleren&ref=card-profile&utm_campaign=affiliate&utm_medium=card&utm_source=scryfall'}};

test('renders correctly', async () => {
  navigationMock.isFocused.mockReturnValueOnce(true);
  jest.useFakeTimers();

  const {toJSON} = await render(<AddCardScan navigation={navigationMock}/>);

  expect(toJSON()).toMatchSnapshot();
});

test('can add card', async () => {
  navigationMock.isFocused.mockReturnValueOnce(true);
  jest.useFakeTimers();

  scryfall.Cards.search.mockImplementation(query => {
    expect(query).toBe('"Jace Beleren" lang:en');
    return {waitForAll: () => Promise.resolve([mockCardResponse])};
  });

  put.mockImplementation(doc => Promise.resolve({...doc, _rev: 'mock-rev'}));

  await render(<AddCardScan navigation={navigationMock}/>);

  await triggerTextRecognition();

  expect(put.mock.calls[0]).toMatchSnapshot();
});
