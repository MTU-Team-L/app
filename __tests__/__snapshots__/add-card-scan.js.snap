// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add card 1`] = `
Array [
  Object {
    "_id": "97956927-135f-47b0-8aef-cc63889eab5a",
    "artist": "Kev Walker",
    "artist_ids": Array [
      "f366a0ee-a0cd-466d-ba6a-90058c7a31a6",
    ],
    "booster": false,
    "border_color": "black",
    "card_back_id": "0aeebaf5-8c7d-4636-9e82-8c27447861f7",
    "cmc": 3,
    "collector_number": "1",
    "color_identity": Array [
      "U",
    ],
    "colors": Array [
      "U",
    ],
    "digital": false,
    "edhrec_rank": 884,
    "foil": true,
    "frame": "2015",
    "full_art": false,
    "games": Array [
      "paper",
    ],
    "highres_image": true,
    "illustration_id": "00146337-02df-4b1e-976b-8d64af966dd8",
    "image_uris": Object {
      "art_crop": "https://img.scryfall.com/cards/art_crop/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464",
      "border_crop": "https://img.scryfall.com/cards/border_crop/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464",
      "large": "https://img.scryfall.com/cards/large/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464",
      "normal": "https://img.scryfall.com/cards/normal/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464",
      "png": "https://img.scryfall.com/cards/png/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.png?1562378464",
      "small": "https://img.scryfall.com/cards/small/front/9/7/97956927-135f-47b0-8aef-cc63889eab5a.jpg?1562378464",
    },
    "lang": "en",
    "layout": "normal",
    "legalities": Object {
      "brawl": "not_legal",
      "commander": "legal",
      "duel": "legal",
      "future": "not_legal",
      "historic": "not_legal",
      "legacy": "legal",
      "modern": "legal",
      "oldschool": "not_legal",
      "pauper": "not_legal",
      "penny": "not_legal",
      "pioneer": "not_legal",
      "standard": "not_legal",
      "vintage": "legal",
    },
    "loyalty": "3",
    "mana_cost": "{1}{U}{U}",
    "multiverse_ids": Array [
      393851,
    ],
    "name": "Jace Beleren",
    "nonfoil": false,
    "object": "card",
    "oracle_id": "cc2b6e58-2d6f-495e-95f3-9496105c4cba",
    "oracle_text": "+2: Each player draws a card.
−1: Target player draws a card.
−10: Target player puts the top twenty cards of their library into their graveyard.",
    "oversized": false,
    "prices": Object {
      "eur": null,
      "tix": null,
      "usd": null,
      "usd_foil": "3.94",
    },
    "prints_search_uri": "https://api.scryfall.com/cards/search?order=released&q=oracleid%3Acc2b6e58-2d6f-495e-95f3-9496105c4cba&unique=prints",
    "promo": false,
    "purchase_uris": Object {
      "cardhoarder": "https://www.cardhoarder.com/cards?affiliate_id=scryfall&data%5Bsearch%5D=Jace+Beleren&ref=card-profile&utm_campaign=affiliate&utm_medium=card&utm_source=scryfall",
      "cardmarket": "https://www.cardmarket.com/en/Magic/Products/Search?referrer=scryfall&searchString=Jace+Beleren&utm_campaign=card_prices&utm_medium=text&utm_source=scryfall",
      "tcgplayer": "https://shop.tcgplayer.com/product/productsearch?id=93796&partner=Scryfall&utm_campaign=affiliate&utm_medium=api&utm_source=scryfall",
    },
    "rarity": "mythic",
    "related_uris": Object {
      "edhrec": "https://edhrec.com/route/?cc=Jace+Beleren",
      "gatherer": "https://gatherer.wizards.com/Pages/Card/Details.aspx?multiverseid=393851",
      "mtgtop8": "https://mtgtop8.com/search?MD_check=1&SB_check=1&cards=Jace+Beleren",
      "tcgplayer_decks": "https://decks.tcgplayer.com/magic/deck/search?contains=Jace+Beleren&page=1&partner=Scryfall&utm_campaign=affiliate&utm_medium=api&utm_source=scryfall",
    },
    "released_at": "2014-12-05",
    "reprint": true,
    "reserved": false,
    "rulings_uri": "https://api.scryfall.com/cards/97956927-135f-47b0-8aef-cc63889eab5a/rulings",
    "scryfall_set_uri": "https://scryfall.com/sets/jvc?utm_source=api",
    "scryfall_uri": "https://scryfall.com/card/jvc/1/jace-beleren?utm_source=api",
    "set": "jvc",
    "set_name": "Duel Decks Anthology: Jace vs. Chandra",
    "set_search_uri": "https://api.scryfall.com/cards/search?order=set&q=e%3Ajvc&unique=prints",
    "set_type": "duel_deck",
    "set_uri": "https://api.scryfall.com/sets/37832684-2fe0-4b06-842b-eec06e5a17cb",
    "story_spotlight": false,
    "tcgplayer_id": 93796,
    "textless": false,
    "type_line": "Legendary Planeswalker — Jace",
    "uri": "https://api.scryfall.com/cards/97956927-135f-47b0-8aef-cc63889eab5a",
    "variation": false,
  },
]
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View />
  <View
    style={
      Object {
        "borderColor": "rgba(0, 255, 13, 1)",
        "borderRadius": 8,
        "borderWidth": 2,
        "height": 324,
        "marginLeft": "auto",
        "marginRight": "auto",
        "position": "absolute",
        "width": 224,
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "rgba(0, 255, 13, 1)",
          "borderRadius": 8,
          "borderWidth": 2,
          "height": 30,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 5,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View />
  <View />
</View>
`;
